### PR TITLE
Add spell checking

### PR DESCRIPTION
**Pull request by danieljurek**
Check spelling on PRs and provide warnings for spelling errors on files changed in the PR. For each file changed in the PR the entire file is scanned (not just the changed sections) and spelling errors are highlighted as warnings in the build. Spelling errors will not block the PR. 

`.vscode/cspell.json` is seeded with exceptions found in a cursory examination of the entire repo.

Example warnings: 
image

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]
Looks good. Lets make sure we get the ci job to run though.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -387,6 +387,8 @@ jobs:
     name: azsdk-pool-mms-win-2019-general
     vmImage: MMS2019
   steps:
+    - template: /eng/common/pipelines/templates/steps/check-spelling.yml
+
```
Comment (ahsonkhan) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:391]: question: is this new line necessary in yml (and has some semantic meaning), or is it just cosmetic?

File: .vscode/cspell.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
+{
+  "version": "0.1",
+  "language": "en",
+  "languageId": "cpp",
```
Comment (ahsonkhan) [inline .vscode/cspell.json:4]: Is there a c specific language id?

File: .vscode/cspell.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    "*.a",
+    "*.lib",
+    "build/**",
+    "*.png",
+    "*.svg",
+    "eng/docs/api/assets/style.css",
+    "eng/common/**/*"
+  ],
+  "ignoreRegExpList": [],
+  "ignoreWords": [
```
Comment (ahsonkhan) [inline .vscode/cspell.json:18]: Are these case sensitive?

File: .vscode/cspell.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    "azsdk",
+    "azuresdkforc",
+    "CMAKE",
+    "cmocka",
+    "Cobertura",
+    "ctest",
+    "DCMAKE",
+    "DCMAKE",
```
Comment (ahsonkhan) [inline .vscode/cspell.json:26]: duplicate words

File: .vscode/cspell.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    "deps",
+    "DLOGGING",
+    "docfx",
+    "DPRECONDITIONS",
+    "DUNIT",
+    "dynsections",
```
Comment (ahsonkhan) [inline .vscode/cspell.json:32]: what's this?

File: .vscode/cspell.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    "eabi",
+    "enumvalues",
```
Comment (ahsonkhan) [inline .vscode/cspell.json:34]: why ignore this?

**Pull request merged**